app/src/common/shared/org/mozilla/vrbrowser/ui/views/library/DownloadsView.java

```diff
@@ -352,14 +334,67 @@ protected void showSortingContextMenu(@NonNull View view) {
         menu.show(UIWidget.REQUEST_FOCUS);
     }
 
+    private Comparator<Download> getSorting(@SortingContextMenuWidget.Order int order) {
+        switch (order) {
+            case SortingContextMenuWidget.SORT_FILENAME_AZ:
+                return mAZFileNameComparator;
+            case SortingContextMenuWidget.SORT_FILENAME_ZA:
+                return mZAFilenameComparator;
+            case SortingContextMenuWidget.SORT_DATE_ASC:
+                return mDownloadDateAscComparator;
+            case SortingContextMenuWidget.SORT_DATE_DESC:
+                return mDownloadDateDescComparator;
+            case SortingContextMenuWidget.SORT_SIZE_ASC:
+                return mDownloadSizeAscComparator;
+            case SortingContextMenuWidget.SORT_SIZE_DESC:
+                return mDownloadSizeDescComparator;
+        }
+
+        return mDownloadIdComparator;
+    }
+
     // DownloadsManager.DownloadsListener
 
-    private Comparator<Download> mAZFileNameComparator = (o1, o2) -> o1.getFilename().compareTo(o2.getFilename());
-    private Comparator<Download> mZAFilenameComparator = (o1, o2) -> o2.getFilename().compareTo(o1.getFilename());
-    private Comparator<Download> mDownloadDateAscComparator = (o1, o2) -> (int)(o1.getLastModified() - o2.getLastModified());
-    private Comparator<Download> mDownloadDateDescComparator = (o1, o2) -> (int)(o2.getLastModified() - o1.getLastModified());
-    private Comparator<Download> mDownloadSizeAscComparator = (o1, o2) -> (int)(o1.getSizeBytes() - o2.getSizeBytes());
-    private Comparator<Download> mDownloadSizeDescComparator = (o1, o2) -> (int)(o2.getSizeBytes() - o1.getSizeBytes());
+    private Comparator<Download> mDownloadIdComparator = (o1, o2) -> (int)(o1.getId() - o2.getId());
+
+    private Comparator<Download> mAZFileNameComparator = (o1, o2) -> {
+        int nameDiff = o1.getFilename().compareTo(o2.getFilename());
+        if (nameDiff == 0) {
+            return  mDownloadIdComparator.compare(o1, o2);
+
+        } else {
+            return nameDiff;
+        }
+    };
+    private Comparator<Download> mZAFilenameComparator = (o1, o2) -> {
```

Comment:
Nit: `mAZFileNameComparator` vs `mZAFilenameComparator` `Name` is only capitalized in one of the names. Can address in a follow up.